cfg/pre-commit-config.yaml

```diff
@@ -7,7 +8,53 @@ repos:
     entry: './src/sh/ensure-shebang.sh'
```

Comment:
This is done now, I believe I covered all of the bases and error messages. It's a bit more convoluted than I thought it would be, so hopefully I didn't miss anything, but I focused on making sure things work in CI rather than locally, because the developer can always figure out how to fix things locally (or we can add it to a README or something)